src/beta/resources/referral.js

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable no-dupe-class-members,no-unused-vars,class-methods-use-this */
+import superagent from 'superagent';
+
+import T from 'proptypes';
+import base from '../../resources/base';
+import { propTypes as userPropTypes } from '../../resources/user';
+import RequestError from '../../errors/request';
+import API from '../../easypost';
+
+export const propTypes = Object.assign({}, userPropTypes, {
+  name: T.string.isRequired,
+  email: T.string.isRequired,
+  phone: T.string.isRequired,
+});
+
+const stripeCreditCardUrl = 'https://api.stripe.com/v1/tokens';
+
+/**
+ * Get an instance of the API client using the referral user's API key.
+ * @param {API} api - The API client to copy.
+ * @param {string }referralApiKey - The referral user's API key.
+ * @returns {API} - An instance of the API client.
+ */
+function getReferralApi(api, referralApiKey) {
+  return API.copyApi(api, {
+    apiKey: referralApiKey,
+  });
+}
+
+/**
+ * Get EasyPost's Stripe API key used to create credit cards on Stripe's servers.
+ * @param {API} api - The API client to use.
+ * @returns {string} - The Stripe API key.
+ */
+async function getEasyPostStripeKey(api) {
+  const response = await api.get('partners/stripe_public_key');
+
+  return response.body.public_key;
+}
+
+/**
+ * Send the credit card details to Stripe to get a Stripe credit card ID.
```

Comment:
```suggestion
 * Send the credit card details to Stripe to get a Stripe credit card token.
```

It should be Stripe credit card token, according to their doc: https://stripe.com/docs/api/tokens/create_card